connectors/tests/test_byoc.py

```diff
@@ -787,6 +756,85 @@ async def fetch_by_id(self, doc_id):
     assert not connector.configuration.is_empty()
 
 
+@pytest.mark.asyncio
+async def test_connector_validate_filtering_not_edited(patch_logger):
+    index = Mock()
+    index.update = AsyncMock()
+    validator = Mock()
+    validator.validate_filtering = AsyncMock()
+
+    connector = Connector(elastic_index=index, doc_source=DOC_SOURCE)
+    await connector.validate_filtering(validator=validator)
+
+    validator.validate_filtering.assert_not_awaited()
+    index.update.assert_not_awaited()
+
+
+@pytest.mark.asyncio
+async def test_connector_validate_filtering_invalid(patch_logger):
+    index = Mock()
+    index.update = AsyncMock()
+    validation_result = FilteringValidationResult(
+        state=FilteringValidationState.INVALID,
+        errors=[FilterValidationError(ids=[1], messages="something wrong")],
+    )
+    validator = Mock()
+    validator.validate_filtering = AsyncMock(return_value=validation_result)
+    expected_validation_update_doc = {
+        "filtering": [
+            {
+                "domain": DEFAULT_DOMAIN,
+                "draft": DRAFT_FILTERING_DEFAULT_DOMAIN
+                | {"validation": validation_result.to_dict()},
+                "active": ACTIVE_FILTERING_DEFAULT_DOMAIN,
+            },
+            FILTERING_OTHER_DOMAIN,
+        ]
+    }
+
+    connector = Connector(
+        elastic_index=index, doc_source=deepcopy(DOC_SOURCE_WITH_EDITED_FILTERING)
+    )
+    await connector.validate_filtering(validator=validator)
+
+    validator.validate_filtering.assert_awaited()
+    index.update.assert_awaited_with(
+        doc_id=CONNECTOR_ID, doc=expected_validation_update_doc
+    )
+
+
+@pytest.mark.asyncio
+async def test_connector_validate_filtering_valid(patch_logger):
```

Comment:
I remember we had the discussion around keeping/removing `patch_logger`. Should we keep it for log messages in tests here? AFAIU it was mainly used to silent log messages during test execution.

Reply:
Yes, had a discussion with @artem-shelkovnikov , will have a separate PR to remove all the unnecessary `patch_logger` in tests.